pyro/distributions/torch.py

```diff
@@ -282,11 +282,28 @@ def expand(self, batch_shape):
             return type(self)(df, loc, scale, validate_args=validate_args)
 
 
-class TransformedDistribution(torch.distributions.TransformedDistribution, TorchDistributionMixin):
+class TransformedDistribution(torch.distributions.TransformedDistribution, TorchDistributionMixin, torch.nn.Module):
```

Comment:
What are the specific methods we would like to inherit from `nn.Module`? I am wondering if it safe to inherit off of `Module` for all `TransformedDistribution` and tying ourselves up in terms of future compatibility. If it is only a couple of methods like `.parameters()` and `.cuda()`, we could provide implementations for these (and use composition when needed to call `.module.cuda()` under the hood, for instance).

Reply:
I think basically all the main ones, like `apply`, `cpu`, `cuda`, `load_state_dict`, `named_parameters`, `parameters`, `state_dict`, `to`. IMHO, I think future compatibility wouldn't be a problem since `nn.Module` is one of the main abstraction mechanisms in PyTorch and it's been around for so long

Reply:
We may need to avoid using `@lazy_property`, or add something to reset all lazy properties when calling `.cpu()`, `.cuda()`, and `.to()`.